rllib/evaluation/collectors/agent_collector.py

```diff
@@ -305,12 +314,24 @@ def add_action_reward_next_obs(self, input_values: Dict[str, TensorType]) -> Non
             ):
                 if should_flatten_action_key:
                     v = flatten_to_single_ndarray(v)
+                # Briefly remove dummy value to add to buffer
+                if k in self.gained_from_inference:
+                    dummy = self.buffers[k][0].pop(-1)
```

Comment:
some note for code optimization nerds like me :) pop(-1) takes O(1) in python lists but pop(0) takes O(n). So we are good here

Reply:
Oh there is so much to optimize here 😄